### PR TITLE
Add In-App Messaging and native push notifications support

### PR DESCRIPTION
**Pull request by cgwyllie**
- Add support for in-app messaging based on updating base SDKs
- Handle all push notification features without requiring any 3rd-party push plugins

### Breaking Changes

- Initialisation now requires a `kumulos.json` config file for native, native Android, and the existing JavaScript initialize code for full feature coverage
- `Kumulos.pushTrackOpen()` has been removed (now handled by the SDK)
- Push registrations & notifications are now handled by the SDK so existing 3rd-party push plugins are no longer required

### Release Checklist

Prepare:

- [x] Detail any breaking changes. Breaking changes require a new major version number

Bump versions in:

- [x] package.json
- [x] plugin.xml
- [x] src/ios/KumulosSDKPlugin.m
- [x] src/android/.../KumulosInitProvider.java

Release:

- [ ] Squash and merge to master
- [ ] Delete branch once merged
- [ ] Create tag from master matching chosen version
- [ ] Fill out release notes
- [ ] Run `npm publish`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,10 +16,24 @@ cordova plugin add cordova-plugin-kumulos-sdk
 Add https://*.kumulos.com to the `Content-Security-Policy` meta tag in your app, for example in `www/index.html`:
 
 ```html
-<meta http-equiv="Content-Security-Policy" content="default-src 'self' data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src 'self' 'unsafe-inline'; media-src *; img-src 'self' data: content:; connect-src 'self' https://*.kumulos.com;">
+<meta
+    http-equiv="Content-Security-Policy"
+    content="default-src 'self' data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src 'self' 'unsafe-inline'; media-src *; img-src 'self' data: content:; connect-src 'self' https://*.kumulos.com;"
```
Comment (azlindsay) [inline README.md]: Does this need iar.app.delivery added to the list of safe addresses or is that considered part of the native code?

Comment (cgwyllie) [reply README.md]: It shouldn't, no, the iar requests are done at the native level. This only covers requests made from JS using fetch (i.e. backend / push subs mgmt etc.).

File: index.d.ts [hunk 2]
```diff
@@ -48,14 +58,36 @@ interface KumulosConfig {
...
+     */
+    inAppDeepLinkPressedHandler?: (data: { [key: string]: any }) => void;
+}
+interface InAppInboxItem {
+    id: number;
+    title: string;
+    subtitle: string;
+    availableFrom: string | '';
+    availableTo: string | '';
+    availableDismissedAt: string | '';
```
Comment (azlindsay) [inline index.d.ts]: is this not just dismissedAt ?

Comment (cgwyllie) [reply index.d.ts]: Why yes it is, updated

**Pull request merged**